trecs/models/bass.py

```diff
@@ -46,57 +49,119 @@ def __init__(self, infection_thresholds=None, verbose=False):
+
+        item_representation: :obj:`numpy.ndarray` or None (optional, default: None)
+            A :math:`|A|\\times|I|` matrix representing the similarity between
+            each item and attribute.
+
+        actual_user_representation: :obj:`numpy.ndarray` or None or \
+                            :class:`~components.users.Users` (optional, default: None)
+            Either a :math:`|U|\\times|T|` matrix representing the real user profiles, where
+            :math:`T` is the number of attributes in the real underlying user profile,
+            or a `Users` object that contains the real user profiles or real
+            user-item scores. This matrix is **not** used for recommendations. This
+            is only kept for measurements and the system is unaware of it.
+
+        actual_item_representation: :obj:`numpy.ndarray` or None (optional, default: None)
+            A :math:`|T|\\times|I|` matrix representing the real user profiles, where
+            :math:`T` is the number of attributes in the real underlying item profile.
+            This matrix is **not** used for recommendations. This
+            is only kept for measurements and the system is unaware of it.
+
+        verbose: bool (optional, default: False)
+            If True, enables verbose mode. Disabled by default.
+
+        num_items_per_iter: int (optional, default: 10)
+            Number of items presented to the user per iteration.
+
+        seed: int, None (optional, default: None)
+            Seed for random generator.
+
+    Attributes
+    -----------
+        Inherited by BaseRecommender: :class:`~models.recommender.BaseRecommender`
+    """
 
     def __init__(  # pylint: disable-all
         self,
-        num_users=100,
-        num_items=1,
+        num_users=None,
+        num_items=None,
         infection_state=None,
         infection_thresholds=None,
         item_representation=None,
         user_representation=None,
-        actual_user_scores=None,
+        actual_user_representation=None,
         actual_item_representation=None,
         probabilistic_recommendations=False,
         verbose=False,
         num_items_per_iter=1,
         seed=None,
         **kwargs
     ):
-        # these are not allowed to be None at the same time
-        if all_none(user_representation, num_users, infection_state):
-            raise ValueError(
-                "user_representation, num_users, and infection_state can't be all None"
-            )
-        if all_none(item_representation, num_items, infection_state):
-            raise ValueError(
-                "item_representation, num_items, and infection_state can't be all None"
-            )
-        if not is_array_valid_or_none(user_representation, ndim=2):
-            raise ValueError("user_representation is invalid")
+        default_num_users = 100
+        default_num_items = 1
+        num_users, num_items = validate_user_item_inputs(
```

Comment:
Here I converted the input checks for `BassModel` to use the `validate_user_item_inputs` method, like all the other models do.